pkg/cluster/cluster.go

```diff
@@ -253,3 +251,20 @@ func setOptionsDefaults(options Options) Options {
 
 	return options
 }
+
+// NewClientFunc allows a user to define how to create a client
+type NewClientFunc func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error)
+
+// DefaultNewClient creates the default caching client
+func DefaultNewClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
```

Comment:
we might want to consider making this options or something in the future -- this is getting a bit unwieldy